drivers/hid/hidparse/parser.h

```diff
@@ -0,0 +1,208 @@
+/*
+ * PROJECT:     ReactOS HID Parser Library
+ * LICENSE:     GPL-3.0-or-later (https://spdx.org/licenses/GPL-3.0-or-later)
+ * FILE:        drivers/hid/hidparse/parser.h
+ * PURPOSE:     HID Parser internal structures
+ * COPYRIGHT:   Copyright (C) Roman Masanin (36927roma@gmail.com) 2022
+ */
+
+#ifndef _HIDPARSER_H_
+#define _HIDPARSER_H_
+
+#include <wdm.h>
+#define _HIDPI_
+#define _HIDPI_NO_FUNCTION_MACROS_
+#include <hidpddi.h>
+#include <preparsed.h>
+
+/* HID report related structures */
+
+typedef enum _ITEM_SIZE
+{
+    ITEM_SIZE_0 = 0, // 0b00
+    ITEM_SIZE_1 = 1, // 0b01
+    ITEM_SIZE_2 = 2, // 0b10
+    ITEM_SIZE_4 = 3, // 0b11
+} ITEM_SIZE;
+
+typedef enum _ITEM_TYPE
+{
+    ITEM_TYPE_MAIN = 0,     // 0b00
+    ITEM_TYPE_GLOBAL = 1,   // 0b01
+    ITEM_TYPE_LOCAL = 2,    // 0b10
+    ITEM_TYPE_RESERVED = 3, // 0b11
+} ITEM_TYPE;
+
+typedef enum _MAIN_ITEM_TYPE
+{
+    MAIN_ITEM_TAG_INPUT = 0x8,          // 0b1000
```

Comment:
I guess we can figure out the binary pattern quite easily ^^ (no need for comments)